service/evaluation/evaluation.go

```diff
@@ -0,0 +1,783 @@
+			opts:   opts,
+		}
+	}
+}
+
+// NewService creates a new Evaluation service
+func NewService(opts ...service.Option[Service]) *Service {
+	var err error
+	s := Service{
+		orchestratorAddress: grpcTarget{
+			target: DefaultOrchestratorAddress,
+		},
+		scheduler: make(map[string]*gocron.Scheduler),
+		wg:        make(map[string]*sync.WaitGroup),
+	}
+
+	// Apply service options
+	for _, o := range opts {
+		o(&s)
+	}
+
+	// Default to an in-memory storage, if nothing was explicitly set
+	if s.storage == nil {
+		s.storage, err = inmemory.NewStorage()
+		if err != nil {
+			log.Errorf("Could not initialize the storage: %v", err)
+		}
+	}
+
+	// Default to an allow-all authorization strategy
+	if s.authz == nil {
+		s.authz = &service.AuthorizationStrategyAllowAll{}
+	}
+
+	return &s
+}
+
+// SetAuthorizer implements UsesAuthorizer
+func (s *Service) SetAuthorizer(auth api.Authorizer) {
+	s.authorizer = auth
+}
+
+// Authorizer implements UsesAuthorizer
+func (s *Service) Authorizer() api.Authorizer {
+	return s.authorizer
+}
+
+// StartEvaluation is a method implementation of the evaluation interface: It periodically starts the evaluation of a cloud service and the given controls_in_scope (e.g., EUCS OPS-13, EUCS OPS-13.2) in the target_of_evaluation. If no inteval time is given, the default value is used.
+func (s *Service) StartEvaluation(_ context.Context, req *evaluation.StartEvaluationRequest) (resp *evaluation.StartEvaluationResponse, err error) {
+	var (
+		parentSchedulerTag string
+		interval           int
+		toe                *orchestrator.TargetOfEvaluation
+		toeTag             string
+	)
+
+	// Validate request
+	err = service.ValidateRequest(req)
+	if err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, "%v", err)
+	}
+
+	// Set the interval to the default value if not set. If the interval is set to 0, the default interval is used.
+	if req.GetInterval() == 0 {
+		interval = defaultInterval
+	} else {
+		interval = int(req.GetInterval())
+	}
+
+	// Get orchestrator client. The orchestrator client is used to retrieve necessary information from the Orchestrator, such as assessment_results, controls or targets_of_evaluation.
+	err = s.initOrchestratorClient()
+	if err != nil {
+		err = fmt.Errorf("could not set orchestrator client: %v", err)
+		log.Error(err)
+		return nil, status.Errorf(codes.Internal, "%s", err)
+	}
+
+	// Get Target of Evaluation. The Target of Evaluation is retrieved to get the controls_in_scope, which are then evaluated.
+	toe, err = s.orchestratorClient.GetTargetOfEvaluation(context.Background(), &orchestrator.GetTargetOfEvaluationRequest{
+		CloudServiceId: req.GetCloudServiceId(),
+		CatalogId:      req.GetCatalogId(),
+	})
+	if err != nil {
+		err = fmt.Errorf("could not get target of evaluation: %v", err)
+		log.Error(err)
+		return nil, status.Errorf(codes.Internal, "%s", err)
+	}
+
+	// toeTag is the tag for the scheduler that contains the scheduler jobs for the specific ToE
```

Comment:
toe tag is maybe not a really suitable name for a variable ;) https://en.wikipedia.org/wiki/Toe_tag